modules/net-firewall-policy/net-global.tf

```diff
@@ -44,11 +44,16 @@ resource "google_compute_network_firewall_policy_rule" "net-global" {
   action                  = local.rules[each.key].action
   description             = local.rules[each.key].description
   direction               = local.rules[each.key].direction
-  security_profile_group  = local.rules[each.key].security_profile_group
   disabled                = local.rules[each.key].disabled
   enable_logging          = local.rules[each.key].enable_logging
   priority                = local.rules[each.key].priority
   target_service_accounts = local.rules[each.key].target_service_accounts
+  tls_inspect             = local.rules[each.key].tls_inspect
+  security_profile_group = (
+    try(var.security_profile_group_ids[local.rules[each.key].security_profile_group], null) == null
```

Comment:
```hcl
try(
  var.security_profile_group_ids[local.rules[each.key].security_profile_group],
  local.rules[each.key].security_profile_group
)
```

or lookup which is better than try

```hcl
lookup(
  var.security_profile_group_ids,
  local.rules[each.key].security_profile_group,
  local.rules[each.key].security_profile_group
)
```

Reply:
I tried lookup before and it doesn't work as the second argument (the lookup key) can never be null, which happens instead in this case. I'll use the try.